.travis.yml

```diff
@@ -4,26 +4,57 @@ os:
   - linux
   - osx
 
+env:
+  - PYTHON_VERSION=3.6
+  - PYTHON_VERSION=3.7
+
+before_install:
+  # Install a supported cmake version (>= 3.10)
+  - |
+    if [ "$TRAVIS_OS_NAME" == "linux" ]; then
+      wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda.sh;
+    else
+      curl -o miniconda.sh  https://repo.continuum.io/miniconda/Miniconda3-latest-MacOSX-x86_64.sh;
+    fi
+  - bash miniconda.sh -b -p $HOME/miniconda
+  - export PATH="$HOME/miniconda/bin:$PATH"
+
+  # Set up the env
+  - conda create -q -n test -y python=$PYTHON_VERSION
+  - source activate test
+
+  # Add the necessary channels
+  - conda config --add channels usgs-astrogeology
+  - conda config --add channels conda-forge
+
 install:
-# Install a supported cmake version (>= 3.10)
-- |
-  if [ "$TRAVIS_OS_NAME" == "linux" ]; then
-    wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda.sh;
-  else
-    curl -o miniconda.sh  https://repo.continuum.io/miniconda/Miniconda3-latest-MacOSX-x86_64.sh;
-  fi
-- bash miniconda.sh -b -p $HOME/miniconda
-- export PATH="$HOME/miniconda/bin:$PATH"
-# Add the channels that conda build is going to use
-- conda config --add channels usgs-astrogeology
-- conda config --add channels conda-forge
 # Setup to do the build 
-- conda install -y -q conda-build anaconda-client
+  - conda env update -n test -f environment.yml
+  - conda install -c conda-forge -y pytest
+  - |
+    if [ "$TRAVIS_OS_NAME" == "linux" ]; then
+      conda install -y -c conda-forge/label/gcc7 python=$PYTHON_VERSION
+      conda install -y gcc_linux-64 gxx_linux-64;
+    fi
 
 script:
-# Build and upload on success
-- if [ "${TRAVIS_PULL_REQUEST}" = "false" ] ;then
-    conda config --set anaconda_upload yes && conda build --token $CONDA_UPLOAD_TOKEN recipe -q;
-  else
-    conda build recipe -q;
-  fi
+  - mkdir build && cd build
+  - cmake .. && make
+  - cd python && python setup.py install && cd ..
+  - |
+    if [ "$TRAVIS_OS_NAME" == "linux" ]; then
+      pytest python/tests/ --lib fixture/libfixturecsm.so;
+    else
+      pytest python/tests/ --lib fixture/libfixturecsm.dylib;
+    fi
```

Comment:
Below, a fixture implementation is created. This block tests using the correct fixture. This is because the csmapi dynamically loads a plugin. In this model, we should be able to swap in any plugin that we want. For dev. purposes though, we want an implementation where all we test is the interface.